pkg/controller/jenkins/configuration/base/reconcile.go

```diff
@@ -480,8 +488,9 @@ func isPodTerminating(pod corev1.Pod) bool {
 	return pod.ObjectMeta.DeletionTimestamp != nil
 }
 
-func (r *ReconcileJenkinsBaseConfiguration) isRecreatePodNeeded(currentJenkinsMasterPod corev1.Pod, userAndPasswordHash string) []string {
+func (r *ReconcileJenkinsBaseConfiguration) isRecreatePodNeeded(currentJenkinsMasterPod corev1.Pod, userAndPasswordHash string) *reason.PodRestart {
```

Comment:
I would use the interface as the return type

Reply:
also the name starting with `is...` would suggest that the return type would be `bool` which is no longer the case,
 my suggestion is `checkForPodRecreation`